examples/logging/README.md

```diff
@@ -0,0 +1,340 @@
+
+## Usage 
+
+Both the `Server` and `Pipeline` interfaces can run with logging.
+
+### Server Usage
+
+The startup command (`deepsparse.server`), accepts an optional YAML configuration file (which contains both logging-specific and general configuration details) via the `--config-file` argument. For example:
+
+```bash
+deepsparse.server --config-file server-config.yaml
+```
+
+The logging is configured as described above. System Logging is defined globally at the `Server` level while Data Logging is defined at the `Endpoint` level. In the example below, we create a `Server` with two `Endpoints` (one with a dense and one with a sparse BERT model).
+
+<details>
+    <summary>Click to see server-config.yaml </summary>
+       
+```yaml
+# config.yaml
+
+loggers:
+     -prometheus
+     -custom_logger
+
+system_logging:
+     deployment_details: off
+     request_details: off
+     prediction_latency: on 
+     dynamic_batch_latency: off
+     # resource_utilization:                    < not specified, so enabled by default
+
+endpoints:
+     - task: question_answering
+       route: /dense/predict
+       model: zoo:nlp/question_answering/bert-base/pytorch/huggingface/squad/base-none
+       batch_size: 1
+       data_logging:
+          - pipeline_inputs
+               mappings:
+                - func: builtins:sequence_length
+                  frequency: 500
+                  target: prometheus
+                  
+                - func: path/to/custom/metrics.py:fn2
+                  # frequency:                  < defaults to every 1000 inferences
+                  # target:                     < defaults to all
+                  
+             - engine_inputs
+                mappings:
+                - func: buildins:identity
+                  frequency: 1000
+                  target: custom_logger
+                  
+             # - engine_outputs                 < nothing logged
+             # - pipeline_outputs               < nothing logged
+             
+     - task: question_answering
+       route: /sparse/predict
+       model: zoo:nlp/question_answering/bert-base/pytorch/huggingface/squad/pruned95_obs_quant-none
+       batch_size: 1
+       # data_logging:                          < nothing logged at this endpoint 
+```
+
+</details>
+
+### Pipeline Usage
+
+The `ManagerLogger` class handles logging and is passed as the `logger` argument to a `Pipeline`. `ManagerLogger` is initialized with the `config` argument, which is a path to a configuration file in the format described above. 
+
+For example, with the QA pipeline:
+
+```python
+from deepsparse import Pipeline
+
+# SparseZoo model stub or path to ONNX file
+model_path = "zoo:nlp/question_answering/bert-base/pytorch/huggingface/squad/12layer_pruned80_quant-none-vnni"
+
+# logger object referencing the local logging config file
+logger = ManagerLogger(config="pipeline-logging-config.yaml")
```

Comment:
This is not supported atm.
In theory, any logger that they provide should work, not only the `Manager(Multi)Logger`.

I do not think that we should allow here the users to specify the loggers, which would later be directly fed to the pipeline. Maybe for simple use cases, they could directly plug here a `PythonLogger`. 

But for good performance / real-life usefulness, they'd need some logger composition, which should be handled by some helper function `logger_from_config(config: str)`

@bfineran